src/test/kotlin/no/nav/syfo/behandler/GetBehandlereSpek.kt

```diff
@@ -0,0 +1,217 @@
+                    dialogmeldingEnabled = true,
+                )
+                database.createBehandlerForArbeidstaker(
+                    behandler = fastlege,
+                    arbeidstakerPersonIdent = UserConstants.ARBEIDSTAKER_FNR,
+                    relasjonstype = BehandlerArbeidstakerRelasjonstype.FASTLEGE,
+                )
+
+                mockFastlege(
+                    personident = UserConstants.FASTLEGE_ANNEN_FNR,
+                    partnerId = PartnerId(1),
+                    fastlegeClientMock = fastlegeClientMock,
+                    partnerinfoClientMock = partnerinfoClientMock,
+                )
+
+                var behandlerDTOList: List<BehandlerDTO>
+                runBlocking {
+                    behandlerDTOList = behandlerService.getBehandlerDTOList(UserConstants.ARBEIDSTAKER_FNR, "token", "callId")
+                }
+
+                behandlerDTOList.size shouldBeEqualTo 1
+                behandlerDTOList[0].type shouldBeEqualTo BehandlerArbeidstakerRelasjonstype.FASTLEGE.name
+                behandlerDTOList[0].fnr shouldBeEqualTo UserConstants.FASTLEGE_ANNEN_FNR.value
+            }
+
+            it("return old fastlege as sykmelder") {
+                val fastlegeAndSykmelder = generateBehandler(
+                    behandlerRef = behandlerRef,
+                    partnerId = PartnerId(1),
+                    dialogmeldingEnabled = true,
+                )
+                database.createBehandlerAndTwoArbeidstakerRelasjoner(
+                    behandler = fastlegeAndSykmelder,
+                    arbeidstakerPersonIdent = UserConstants.ARBEIDSTAKER_FNR,
+                    otherArbeidstakerPersonIdent = UserConstants.ARBEIDSTAKER_FNR,
+                    relasjonstype = BehandlerArbeidstakerRelasjonstype.FASTLEGE,
+                    otherRelasjonstype = BehandlerArbeidstakerRelasjonstype.SYKMELDER,
+                )
+                mockFastlege(
+                    personident = UserConstants.FASTLEGE_ANNEN_FNR,
+                    partnerId = PartnerId(1),
+                    fastlegeClientMock = fastlegeClientMock,
+                    partnerinfoClientMock = partnerinfoClientMock,
+                )
+
+                var behandlerDTOList: List<BehandlerDTO>
+                runBlocking {
+                    behandlerDTOList = behandlerService.getBehandlerDTOList(UserConstants.ARBEIDSTAKER_FNR, "token", "callId")
+                }
+
+                behandlerDTOList.size shouldBeEqualTo 2
+                val fastlege = behandlerDTOList.firstOrNull { it.type == BehandlerArbeidstakerRelasjonstype.FASTLEGE.name }
+                fastlege!!.fnr shouldBeEqualTo UserConstants.FASTLEGE_ANNEN_FNR.value
+                val sykmelder = behandlerDTOList.firstOrNull { it.type == BehandlerArbeidstakerRelasjonstype.SYKMELDER.name }
+                sykmelder!!.fnr shouldBeEqualTo UserConstants.FASTLEGE_FNR.value
+            }
+        }
+    }
+})
+
+fun mockFastlege(
```

Comment:
Hvorfor ikke bruke `FastlegeRestMock` og `SyfopartnerInfoMock`?

Reply:
Da må vi vel starte opp hele applikasjonen? 🤔

Reply:
Kunne vel brukt TestApplicationEngine og testet vi APIet ja, men for å teste spesiell logikk i servicen tenker jeg det er greit å gjøre det sånn.

Reply:
@JMLindseth Ser ut som den allerede er startet lenger oppe her:
`val externalMockEnvironment = ExternalMockEnvironment.instance`

Reply:
Vi merger denne nå, så fortsetter vi å se på testene i ny branch, så har vi gjort noe i dag 😄